text/000-merge-with-rubygems.md

```diff
@@ -0,0 +1,100 @@
+
+As part of this transition, Bundler 4 (as part of RubyGems 4) will be able to be able to install any existing Bundler project: full backwards compatibility.
+
+To use the new, merged Bundler and RubyGems, install the latest RubyGems by running `gem update --system`. You're done! You now have the latest Bundler and the latest RubyGems.
+
+From time to time, Bundler may introduce opt-in changes that require a certain minimum version of Bundler. In that case, it is possible to choose to upgrade an application to require the latest version of Bundler in order to use those features.
+
+However, Bundler will never force an application (or other users of an application) to upgrade to the latest Bundler version. The minimum required version of Bundler will always be opt-in, per application, and never be systemwide across an entire Ruby.
+
+## Reference-level explanation
+
+Implementing the merger will likely require multiple phases. The organizational merger is easy to start right away, while the code-level merger will require some research and experimentation beforehand.
+
+### Phase 1, organization merger (April 2019–May 2019)
+- Move GitHub repos from the `bundler/` org to the `rubygems/` org
+- Add all members of the Bundler maintainers team to the existing RubyGems maintainers team to create a new combined maintainers team
+- Combine maintainer Slack workspaces and Slack channels
+- Combine security@, team@, and other emails
+- Combine READMEs, issue reporting guides, and other in-repo docs
+- Update documentation and website links to issues, pull requests, troubleshooting, developer docs, etc
+
+### Phase 2, research for codebase merger (June 2019–??? 2019)
+- Research and test a merge plan that replaces the Bundler submodule in RubyGems with the entire git history and all commits from the Bundler git repo
+- Create a CI task that runs run all of the RubyGems and Bundler tests together, covering the current RubyGems and Bundler build matrix (we can delete SO MANY MATRIX ENTRIES 🤩)
+- Update the Bundler release scripts and automation to release out of the RubyGems repo, and test it
+
+### Phase 3, codebase merger (??? 2019–??? 2020)
+- Use the plan from Phase 2 to merge the Bundler repo into the RubyGems repo in place of the submodule. It's ok if this needs a force push, but we want to avoid more than one force push.
+- Ensure that CI continues to pass with Bundler development moved into the RubyGems repo
+- Use GitHub's "Move Issue" beta feature to move active issues from the Bundler repo over to the RubyGems repo
+- Update the `rubygems/bundler` readme and "new issue" message to point to `rubygems/rubygems` for everything except bug fixes for old Bundler versions
+- Write and test a release script that automatically releases both RubyGems and Bundler together with one version number, one changelog, one tag, and one stable branch, as an artifact that can be installed via `gem install --system`
```

Comment:
how will rubygems handle different versions of the bundler gem being installed / attempted to be used? 

how will bundler handle attempts to be run with a different version of rubygems?

Reply:
> how will bundler handle attempts to be run with a different version of rubygems?

Is it in the test environment? The rubygems never work the multiple versions with the single ruby version in the real-world usage.

Reply:
I meant, what will happen if someone installs a bundler using a different version of rubygems than we intend to support

Reply:
Bundler and RubyGems fully combining means you will only be able to install one copy of both at the same time. To be clear that is a future, eventual goal: the immediate goal is to get them both shipping from the same repo with their own separate versions and separate packages and separate installation.

Later, after releasing both packages out of one repo is working, I am proposing that we plan to combine releases, with one single "RubyGems/Bundler" thing that can only be installed as one blob, and that can only have one version per Ruby.